### PR TITLE
Update to ODF schema changes

### PR DESCRIPTION
**Pull request by sergiimk**
This PR reflects changes proposed in open-data-fabric/open-data-fabric#71, most notably:
- New enum tagging scheme in YAML per ODF RFC-013
- New offset and block range propagation scheme per ODF RFC-014
  - This significantly simplified logic in some places!
- A much simpler structure of `TransformInput` containing `dataset_ref` (ID) and an `alias` (string for queries only)
- Some other renaming and format changes
  - Most notably a switch to `u64` sequence numbers

### CODE REVIEW

**Comment by sergiimk** [comment]
I recommend reviewing individual commits if you want to separate the core changes from the noise of updating the tests.

File: src/adapter/http/src/smart_protocol/protocol_dataset_helper.rs [hunk 1]
```diff
@@ -220,15 +220,12 @@ pub async fn dataset_append_metadata(
...
                 } else {
-                    return Err(AppendError::InvalidBlock(
-                        AppendValidationError::InvalidEvent(InvalidEventError::new(
-                            block.event,
-                            "Transform input with unresolved ID",
-                        )),
-                    ));
+                    // Input references must be resolved to IDs here, but we
+                    // ignore the errors and let the metadata chain reject this
+                    // event
```
Comment (s373r) [inline src/adapter/http/src/smart_protocol/protocol_dataset_helper.rs:228]: Previously we had:
```rust
                    return Err(AppendError::InvalidBlock(
                        AppendValidationError::InvalidEvent(InvalidEventError::new(
                            block.event,
                            "Transform input with unresolved ID",
                        )),
                    ));
```

Q: Could you please clarify: how will the event be rejected?

File: src/infra/core/src/repos/dataset_impl.rs [hunk 2]
```diff
@@ -414,16 +421,13 @@ where
...
                 } else {
-                    return Err(CommitError::MetadataAppendError(AppendError::InvalidBlock(
-                        AppendValidationError::InvalidEvent(InvalidEventError::new(
-                            event,
-                            "Transform input with unresolved ID",
-                        )),
-                    )));
+                    // Normally all references must be resolved to IDs already.
+                    // We continue here while expecting MetadataChain to reject
+                    // this event.
```
Comment (s373r) [inline src/infra/core/src/repos/dataset_impl.rs:430]: Previously we had:
```rust
return Err(CommitError::MetadataAppendError(AppendError::InvalidBlock(
    AppendValidationError::InvalidEvent(InvalidEventError::new(
        event,
        "Transform input with unresolved ID",
    )),
)));
```
**Q:** Could you please clarify: how will the event be rejected?

File: src/infra/core/tests/tests/repos/test_metadata_chain_impl.rs [hunk 3]
```diff
@@ -209,18 +209,25 @@ async fn test_append_prev_block_sequence_integrity_broken() {
...
-    let block_too_low =
-        MetadataFactory::metadata_block(MetadataFactory::add_data().interval(10, 19).build())
-            .prev(&hash, 1 /* should be 2 */)
-            .build();
+    let block_too_low = MetadataFactory::metadata_block(
+        MetadataFactory::add_data()
+            .new_offset_interval(10, 19)
+            .build(),
+    )
+    .prev(&hash, 1 /* should be 2 */)
```
Comment (s373r) [inline src/infra/core/tests/tests/repos/test_metadata_chain_impl.rs:229]: please add a comment w/ context why it should be

File: src/infra/core/tests/tests/repos/test_metadata_chain_impl.rs [hunk 4]
```diff
@@ -234,10 +241,13 @@ async fn test_append_prev_block_sequence_integrity_broken() {
...
-    let block_too_high =
-        MetadataFactory::metadata_block(MetadataFactory::add_data().interval(10, 19).build())
-            .prev(&hash, 3 /* should be 2 */)
-            .build();
+    let block_too_high = MetadataFactory::metadata_block(
+        MetadataFactory::add_data()
+            .new_offset_interval(10, 19)
+            .build(),
+    )
+    .prev(&hash, 3 /* should be 2 */)
```
Comment (s373r) [inline src/infra/core/tests/tests/repos/test_metadata_chain_impl.rs:249]: please add a comment w/ context why it should be

**Review by s373r — COMMENTED** [review]
Added several questions and minor comments

File: src/adapter/http/src/smart_protocol/protocol_dataset_helper.rs (hunk 1 above)
Comment (sergiimk) [reply src/adapter/http/src/smart_protocol/protocol_dataset_helper.rs]: See `metadata_chain_impl.rs` - it contains a bunch of validation logic and is our "ground source of truth" on what is considered a valid metadata.

Ideally we want to make states like this unrepresentable in the type system and I already have a ticket for that:
https://github.com/open-data-fabric/open-data-fabric/issues/73

File: src/infra/core/tests/tests/repos/test_metadata_chain_impl.rs (hunk 3 above)
Comment (sergiimk) [reply src/infra/core/tests/tests/repos/test_metadata_chain_impl.rs]: I think the code here is fairly self-documenting:
- the test name is about block sequence integrity
- right below we assert on sequence integrity error

**Pull request merged**